yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/YamlTestExtension.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * YamlTestExtension.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.relational.yamltests;
+
+import com.apple.foundationdb.relational.yamltests.configs.EmbeddedConfig;
+import com.apple.foundationdb.relational.yamltests.configs.JDBCInProcessConfig;
+import com.apple.foundationdb.relational.yamltests.configs.MultiServerConfig;
+import com.apple.foundationdb.relational.yamltests.configs.YamlTestConfig;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContext;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContextProvider;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+/**
+ * Extension that backs {@link YamlTest}.
+ */
+public class YamlTestExtension implements TestTemplateInvocationContextProvider, BeforeAllCallback, AfterAllCallback {
+    private static final Logger logger = LogManager.getLogger(YamlTestExtension.class);
+    private List<YamlTestConfig> testConfigs;
+
+    @Override
+    public void beforeAll(final ExtensionContext context) throws Exception {
+        if (Boolean.parseBoolean(System.getProperty("tests.runQuick", "false"))) {
+            testConfigs = List.of(new EmbeddedConfig());
+        } else {
+            testConfigs = List.of(
+                    new EmbeddedConfig(),
+                    new JDBCInProcessConfig(),
+                    new MultiServerConfig(0, 1111, 1112),
+                    new MultiServerConfig(1, 1113, 1114)
+            );
+        }
+        for (final YamlTestConfig testConfig : testConfigs) {
+            testConfig.beforeAll();
```

Comment:
Would be nice to find a way to run the `BeforeAll` prior to execution each config, to avoid cross-config conflicts.